components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java

```diff
@@ -1230,14 +1232,12 @@ protected String getIssuer(AuthenticationContext context) {
 
     private EncryptedKey getEncryptedKey(EncryptedAssertion encryptedAssertion) throws Exception {
 
-        try {
-            if (encryptedAssertion.getEncryptedData().getKeyInfo().getEncryptedKeys().isEmpty()) {
-                return encryptedAssertion.getEncryptedKeys().get(0);
-            } else {
-                return encryptedAssertion.getEncryptedData().getKeyInfo().getEncryptedKeys().get(0);
-            }
-        } catch (Exception e) {
-            throw new Exception(" Can not get the encrypted key ", e);
+        if (isNotEmpty(encryptedAssertion.getEncryptedData().getKeyInfo().getEncryptedKeys())) {
+            return encryptedAssertion.getEncryptedData().getKeyInfo().getEncryptedKeys().get(0);
```

Comment:
Let's have some debug logs to indicate where we got the key from
eg: <KeyInfo> element or Assertion

Reply:
fixed